### ISSUE TITLE
possible memory leak

### ISSUE THREAD

**Issue by StrawberryFlavor** [AcalaNetwork/chopsticks#189]
node version : `v18.12.1`
```
<--- Last few GCs --->
2)[2840095:0x52eea70]  1332541 ms: Mark-sweep (reduce) 1714.4 (1770.8) -> 1714.4 (1770.8) MB, 833.5 / 0.0 ms  (+ 0.3 ms in 5 steps since start of marking, biggest step 0.2 ms, walltime since start of marking 839 ms) (average mu = 0.272, current mu = 0.243) [2840095:0x52eea70]  1333641 ms: Mark-sweep (reduce) 1714.4 (1770.8) -> 1714.4 (1770.8) MB, 832.4 / 0.0 ms  (+ 0.2 ms in 1 steps since start of marking, biggest step 0.2 ms, walltime since start of marking 837 ms) (average mu = 0.258, current mu = 0.243) 

<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0xb6e500 node::Abort() [/usr/local/bin/node]
 2: 0xa7e632  [/usr/local/bin/node]
 3: 0xd47f20 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 4: 0xd482c7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 5: 0xf25685  [/usr/local/bin/node]
 6: 0xf26588 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [/usr/local/bin/node]
 7: 0xf36a93  [/usr/local/bin/node]
 8: 0xf37908 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/local/bin/node]
 9: 0xf38a29 v8::internal::Heap::ReportExternalMemoryPressure() [/usr/local/bin/node]
10: 0xd5b692 v8::Isolate::AdjustAmountOfExternalAllocatedMemory(long) [/usr/local/bin/node]
11: 0x10f28fc v8::internal::JSArrayBuffer::Setup(v8::internal::SharedFlag, v8::internal::ResizableFlag, std::shared_ptr<v8::internal::BackingStore>) [/usr/local/bin/node]
12: 0xf000c2 v8::internal::Factory::NewJSArrayBuffer(std::shared_ptr<v8::internal::BackingStore>, v8::internal::AllocationType) [/usr/local/bin/node]
13: 0x154275e v8::internal::WasmMemoryObject::Grow(v8::internal::Isolate*, v8::internal::Handle<v8::internal::WasmMemoryObject>, unsigned int) [/usr/local/bin/node]
14: 0x1403d6e v8::internal::Runtime_WasmMemoryGrow(int, unsigned long*, v8::internal::Isolate*) [/usr/local/bin/node]
15: 0x16e99f9  [/usr/local/bin/node]
```

![image](https://user-images.githubusercontent.com/28721839/216812394-4c0fd8ff-993e-4044-b6de-d2e382ee6cec.png)
<img width="996" alt="image" src="https://user-images.githubusercontent.com/28721839/216812401-988b10d0-45fd-4956-a82d-b7a14ebca285.png">

**Comment by DrW3RK** [comment]
I have the same error.

<img width="1382" alt="image" src="https://user-images.githubusercontent.com/86818441/219432313-306caebb-e435-48ae-a00f-e0ffe1244366.png">

**Comment by xlc** [comment]
Yeah right now Chopstick to run as a long lasting server. We need to implement block offloading to database so it don’t save blocks in memory forever.

**Comment by DrW3RK** [comment]
Thanks @xlc ! What would be the best way to test out features that require waiting for some time, like OpenGov proposals and fellowship member voting etc. using Chopsticks? I saw an RPC call for `dev_timeTravel` but could not get that to work. Any input is appreciated.

**Comment by xlc** [comment]
Yeah it is a bit hard. It is possible to implement a time wrap feature that jump block but that could introduce inconsistency if not used correctly. 

Another way will be directly modify scheduled items in storage to make then happen sooner.

**Comment by ermalkaleci** [comment]
we've also added support to dry-run preimage you can give it a try

**Comment by xlc** [comment]
will be fixed by #347
